vector/src/main/java/im/vector/adapters/RoomDirectoryAdapter.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2017 Vector Creations Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.adapters;
+
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.graphics.drawable.Drawable;
+import android.os.AsyncTask;
+import android.support.v7.widget.RecyclerView;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Filter;
+import android.widget.Filterable;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import org.matrix.androidsdk.util.Log;
+
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import im.vector.R;
+import im.vector.util.RoomDirectoryData;
+
+public class RoomDirectoryAdapter extends RecyclerView.Adapter<RoomDirectoryAdapter.RoomDirectoryViewHolder> implements Filterable {
+
+    private static final String LOG_TAG = "RoomDirectoryAdapter";
+
+    private List<RoomDirectoryData> mList;
+    private List<RoomDirectoryData> mFilteredList;
+
+    private final OnSelectRoomDirectoryListener mListener;
+
+    /*
+     * *********************************************************************************************
+     * Constructor
+     * *********************************************************************************************
+     */
+
+    public RoomDirectoryAdapter(final List<RoomDirectoryData> serversList, final OnSelectRoomDirectoryListener listener) {
+        mList = serversList;
+        mFilteredList = new ArrayList<>(serversList);
+        mListener = listener;
+    }
+
+    /**
+     * Update the servers list
+     * @param serversList the new servers list
+     */
+    public void updateDirectoryServersList(List<RoomDirectoryData> serversList) {
+        mList = serversList;
```

Comment:
For more safety, maybe `clear()` then `addAll(serversList)` instead of assigning a new list?